kratos/includes/model_part.h

```diff
@@ -1353,7 +1353,11 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
 
     ModelPart* GetParentModelPart() const
     {
-        return mpParentModelPart;
+        if (IsSubModelPart()) {
+            return mpParentModelPart;
```

Comment:
It would be better to fill the parent modelpart correctly and just returning it here

Reply:
+1 for this suggestion, I guess this would also fix other places that have the same problem